src/h5web/vis-packs/core/shared/Html.tsx

```diff
@@ -4,38 +4,82 @@ import {
   forwardRef,
   HTMLAttributes,
   useLayoutEffect,
+  useRef,
+  PropsWithChildren,
+  ForwardedRef,
 } from 'react';
 import ReactDOM from 'react-dom';
-import { useThree } from '@react-three/fiber';
+import { useFrame, useThree } from '@react-three/fiber';
+import { Camera, Group, Object3D, Vector3 } from 'three';
 
 // Simplified version of `drei`'s `<Html>` component
-// https://github.com/pmndrs/drei/blob/v2.2.3/src/Html.tsx
-const Html = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
-  (props, ref) => {
-    const { className, style, children, ...divProps } = props;
+// https://github.com/pmndrs/drei/blob/v6.0.3/src/web/Html.tsx
 
-    const { width, height } = useThree((state) => state.size);
+const v1 = new Vector3();
+
+function defaultCalculatePosition(
+  el: Object3D,
+  camera: Camera,
+  size: { width: number; height: number }
+) {
+  const objectPos = v1.setFromMatrixPosition(el.matrixWorld);
+  objectPos.project(camera);
+  const widthHalf = size.width / 2;
+  const heightHalf = size.height / 2;
+  return [
+    objectPos.x * widthHalf + widthHalf,
+    -(objectPos.y * heightHalf) + heightHalf,
+  ];
+}
+
+export type CalculatePosition = typeof defaultCalculatePosition;
+
+export interface HtmlProps
+  extends PropsWithChildren<HTMLAttributes<HTMLDivElement>> {
+  calculatePosition?: CalculatePosition;
+}
+
+const Html = forwardRef(
+  (props: HtmlProps, ref: ForwardedRef<HTMLDivElement>) => {
+    const {
+      className,
+      style,
+      children,
+      calculatePosition = defaultCalculatePosition,
+      ...divProps
+    } = props;
+
+    const size = useThree((state) => state.size);
     const gl = useThree((state) => state.gl);
-    const { parentElement } = gl.domElement;
+    const { parentNode } = gl.domElement;
+    const scene = useThree(({ scene }) => scene);
+    const camera = useThree(({ camera }) => camera);
 
     // Container `div` for ReactDOM to render into, appended next to R3F's `canvas`
     const [el] = useState(() => document.createElement('div'));
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const group = useRef<Group>(null!);
 
+    // eslint-disable-next-line consistent-return
     useEffect(() => {
-      el.style.cssText = `position: absolute; top: 0; left: 0;`;
+      if (group.current) {
+        scene.updateMatrixWorld();
```

Comment:
Do you think this is needed in our app? Same with `camera.updateMatrixWorld()` below.

Reply:
Removed it